backend/server/src/db/model/eval.rs

```diff
@@ -0,0 +1,111 @@
+use super::ForInsert;
+use crate::db::DbService;
+use sqlx::FromRow;
+use std::collections::HashMap;
+use tracing::debug;
+
+#[derive(Clone, Debug, FromRow)]
+pub struct Drv {
+    pub id: i64,
+
+    /// Derivation store path
+    pub drv_path: String,
+
+    /// to reattempt the build (depending on the interruption kind).
+    pub system: String,
+}
+
+impl Drv {
+    /// Create a partially filled row
+    pub fn for_insert(drv_path: String, system: String) -> ForInsert<Self> {
+        ForInsert(Self {
+            id: 1,
+            drv_path,
+            system,
+        })
+    }
+}
+
+impl DbService {
+    pub async fn has_drv(&self, drv_path: &str) -> anyhow::Result<bool> {
+        let result = sqlx::query_scalar("SELECT EXISTS(SELECT 1 FROM Drv WHERE drv_path = $1)")
+            .bind(drv_path)
+            .fetch_one(&self.pool)
+            .await?;
+        Ok(result)
+    }
+
+    /// This will insert a hashmap of <drv, Vec<referrences>> into
+    /// the database. The assumption is that the keys are new drvs and the
+    /// references may or may not already exist
+    pub async fn insert_drv_graph(
```

Comment:
I think it is not a given that the keys are not yet in the database, maybe use a `ON CONFLICT IGNORE` for the drv inserts?

Reply:
I thought the same thing when moving over Drv table. Looks like I forgot to add that. Thanks

Reply:
added